spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :request do
+  describe 'Get id' do
+    it 'returns a successful response' do
+      get '/users/:id/posts/:id/'
+      expect(response).to be_successful
+    end
+
```

Comment:
- [ ] Great job so far 👏🏻 However the `index` and `show` actions for `User` and `Post` controllers should be tested. Actually, you're missing the `show` actions tests. please add them and make sure to include tests that ensure the response body includes the correct placeholder text. Example : `expect(response.body).to include("Welcome to user index page")`

```ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
```